synop2bufr/__init__.py

```diff
@@ -1224,44 +1222,74 @@ def transform(data: str, metadata: str, year: int,
                 extended_delayed_replications,
                 table_version)
         except Exception as e:
+            LOGGER.error(e)
             LOGGER.error("Error creating BUFRMessage")
-            raise e
+            conversion_success[tsi] = False
 
         # parse
-        message.parse(msg, mapping)
-
-        try:
-            result["bufr4"] = message.as_bufr()  # encode to BUFR
-        except Exception as e:
-            LOGGER.error("Error encoding BUFR, null returned")
-            LOGGER.error(e)
-            result["bufr4"] = None
-
-        # now identifier based on WSI and observation date as identifier
-        isodate = message.get_datetime().strftime('%Y%m%dT%H%M%S')
-        rmk = f"WIGOS_{wsi}_{isodate}"
-
-        # now additional metadata elements
-        result["_meta"] = {
-            "id": rmk,
-            "geometry": {
-                "type": "Point",
-                "coordinates": [
-                    message.get_element('#1#longitude'),
-                    message.get_element('#1#latitude')
-                ]
-            },
-            "properties": {
-                "md5": message.md5(),
-                "wigos_station_identifier": wsi,
-                "datetime": message.get_datetime(),
-                "originating_centre": message.get_element("bufrHeaderCentre"),
-                "data_category": message.get_element("dataCategory")
+        if conversion_success[tsi]:
+            try:
+                message.parse(msg, mapping)
+            except Exception as e:
+                LOGGER.error(e)
+                LOGGER.error("Error parsing message")
+                conversion_success[tsi] = False
+
+        # Only convert to BUFR if there's no errors so far
+        if conversion_success[tsi]:
```

Comment:
Even if unsuccessful we need to yield a result, I'm not sure that this is not always the case looking at the above code.

For consistency with the other BUFR tools being developed the status should be included in the _meta object, e.g. 

```python
result['_meta']['status'] = {
   'code':  SUCCESS|FAILED
   'message': 'some high level error / status message'
   'errors': ['array of errors', 'error 1', 'error 2', 'etc]
}

```
where `SUCCESS` and `FAILED` are enumerated.